File: backend/productApi/src/main/java/org/acme/Product.java
package org.acme;

import io.quarkus.hibernate.orm.panache.PanacheEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;

@Entity
@Table(name = "products")
public class Product extends PanacheEntity {
    @Column(length = 100)
    public String name;

    @Column(length = 100)
    public String brand;

    public int portionSizeInGrams;
    public int caloriesPer100Grams;
    public double proteinsInGrams;
    public double fatsInGrams;
    public double carbsInGrams;
}

File: backend/tracker/infrastructure/entity/product.go
package entity

type Product struct {
	ID                  string  `json:"ID" `
	Name                string  `json:"Name"`
	Brand               string  `json:"Brand"`
	PortionSizeInGrams  int     `json:"PortionSizeInGrams"`
	CaloriesPer100Grams int     `json:"CaloriesPer100Grams"`
	ProteinsInGrams     float64 `json:"ProteinsInGrams"`
	FatsInGrams         float64 `json:"FatsInGrams"`
	CarbsInGrams        float64 `json:"CarbsInGrams"`
}
